app/main.py

```diff
@@ -1 +1,53 @@
-# write your code here
+class Animal:
+    alive = []  # Класс-атрибут для хранения всех живых животных
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        """
+        Инициализирует животное с именем, здоровьем и скрытностью.
+        """
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        """
+        Возвращает строковое представление животного.
+        """
+        return (
+            f"{{Name: {self.name}, Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def die(self) -> None:
+        """
+        Удаляет животное из списка живых, если здоровье равно или меньше 0.
+        """
+        if self in Animal.alive:
+            Animal.alive.remove(self)
```

Comment:
The `die` method should check if the animal's health is less than or equal to 0 before removing it from the `alive` list. Currently, it only checks if the animal is in the list, which does not align with the method's docstring stating it should remove the animal if health is 0 or less.